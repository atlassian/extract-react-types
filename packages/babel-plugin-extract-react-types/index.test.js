const { transformSync } = require('@babel/core');
const jestInCase = require('jest-in-case');

let flowCases = [
  {
    name: 'class',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };

    export class SomeComponent extends React.Component<Props> {
      render() {
        return null;
      }
    }
    `
  },
  {
    name: 'arrow function',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };
    
    export const SomeComponent = (props: Props) => {
      return null;
    };  
    `
  },
  {
    name: 'function declaration',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };
    
    export function SomeComponent(props: Props) {
      return null;
    }
    `
  },
  {
    name: 'default class',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };

    export default class SomeComponent extends React.Component<Props> {
      render() {
        return null;
      }
    }
    `
  },
  {
    name: 'default function declaration',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };

    export default class SomeComponent extends React.Component<Props> {
      render() {
        return null;
      }
    }
    `
  },
  {
    name: 'class declaration export default identifier',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };

    class SomeComponent extends React.Component<Props> {
      render() {
        return null;
      }
    }

    export default SomeComponent
    `
  },
  {
    name: 'arrow function export default identifier',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };
    
    const SomeComponent = (props: Props) => {
      return null;
    };  

    export default SomeComponent
    `
  },
  {
    name: 'export default function declaration',
    code: `
    // @flow

    type Props = {
      /** this does something */
      wow: boolean
    };
    
    export default function SomeComponent (props: Props) {
      return null;
    };  
    `
  },
  {
    name: 'React.memo',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = React.memo((props: Props) => {
        return null
      })
      `
  },
  {
    name: 'memo',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = memo((props: Props) => {
        return null
      })
      `
  },
  {
    name: 'forwardRef',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = forwardRef((props: Props) => {
        return null
      })
      `
  },
  {
    name: 'forwardRef function expression',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = forwardRef(function (props: Props) {
        return null
      })
      `
  },
  {
    name: 'forwardRef memo function expression',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = memo(forwardRef(function (props: Props) {
        return null
      }))
      `
  },
  {
    name: 'forwardRef memo arrow function',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      export const SomeComponent = memo(forwardRef((props: Props) => {
        return null
      }))
      `
  },
  {
    name: 'arrow function then export',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      const SomeComponent = (props: Props) => {
        return null
      }

      export { SomeComponent }
      `
  },
  {
    name: 'class declaration then export',
    code: `
      // @flow
  
      type Props = {
        /** this does something */
        wow: boolean
      };
      
      class SomeComponent extends React.Component<Props> {
        render() {
          return null;
        }
      }

      export { SomeComponent }
      `
  }
];

jestInCase(
  'flow',
  ({ code }) => {
    let transformedCode = transformSync(code, {
      plugins: [
        require.resolve('./index'),
        require.resolve('@babel/plugin-syntax-jsx'),
        require.resolve('@babel/plugin-syntax-flow')
      ],
      babelrc: false,
      configFile: false
    }).code;
    expect(transformedCode).toMatchSnapshot();
  },
  flowCases
);

let tsCases = [
  {
    name: 'named',
    code: `export class SomeComponent extends React.Component<{ foo: boolean }> {

    }`
  },
  {
    name: 'default',
    code: `export default class SomeComponent extends React.Component<{ foo: boolean }> {

    }`
  }
];

jestInCase(
  'typescript',
  ({ code }) => {
    let transformedCode = transformSync(code, {
      plugins: [
        require.resolve('./index'),
        require.resolve('@babel/plugin-syntax-jsx'),
        require.resolve('@babel/plugin-syntax-typescript')
      ],
      babelrc: false,
      configFile: false
    }).code;
    expect(transformedCode).toMatchSnapshot();
  },
  tsCases
);
