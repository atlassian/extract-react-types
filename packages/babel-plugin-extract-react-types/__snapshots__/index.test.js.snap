// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flow React.memo 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = React.memo((props: Props) => {
  return null;
});
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow arrow function 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = (props: Props) => {
  return null;
};
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow arrow function export default identifier 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};

const SomeComponent = (props: Props) => {
  return null;
};

export default SomeComponent;
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow arrow function then export 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};

const SomeComponent = (props: Props) => {
  return null;
};

export { SomeComponent };
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow class 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export class SomeComponent extends React.Component<Props> {
  render() {
    return null;
  }

}
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow class declaration export default identifier 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};

class SomeComponent extends React.Component<Props> {
  render() {
    return null;
  }

}

export default SomeComponent;
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow class declaration then export 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};

class SomeComponent extends React.Component<Props> {
  render() {
    return null;
  }

}

export { SomeComponent };
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow default class 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export default class SomeComponent extends React.Component<Props> {
  render() {
    return null;
  }

}
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow default function declaration 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export default class SomeComponent extends React.Component<Props> {
  render() {
    return null;
  }

}
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow export default function declaration 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export default function SomeComponent(props: Props) {
  return null;
}
;
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow forwardRef 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = forwardRef((props: Props) => {
  return null;
});
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow forwardRef function expression 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = forwardRef(function (props: Props) {
  return null;
});
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow forwardRef memo arrow function 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = memo(forwardRef((props: Props) => {
  return null;
}));
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow forwardRef memo function expression 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = memo(forwardRef(function (props: Props) {
  return null;
}));
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow function declaration 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export function SomeComponent(props: Props) {
  return null;
}
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`flow memo 1`] = `
"// @flow
type Props = {
  /** this does something */
  wow: boolean
};
export const SomeComponent = memo((props: Props) => {
  return null;
});
SomeComponent.___types = {
  \\"kind\\": \\"generic\\",
  \\"value\\": {
    \\"kind\\": \\"object\\",
    \\"members\\": [{
      \\"kind\\": \\"property\\",
      \\"key\\": {
        \\"kind\\": \\"id\\",
        \\"name\\": \\"wow\\"
      },
      \\"value\\": {
        \\"kind\\": \\"boolean\\"
      },
      \\"optional\\": false,
      \\"leadingComments\\": [{
        \\"type\\": \\"commentBlock\\",
        \\"value\\": \\"this does something\\",
        \\"raw\\": \\"* this does something \\"
      }]
    }],
    \\"leadingComments\\": [{
      \\"type\\": \\"commentLine\\",
      \\"value\\": \\"@flow\\",
      \\"raw\\": \\" @flow\\"
    }],
    \\"referenceIdName\\": \\"Props\\"
  },
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`typescript default 1`] = `
"export default class SomeComponent extends React.Component<{
  foo: boolean;
}> {}
SomeComponent.___types = {
  \\"kind\\": \\"object\\",
  \\"members\\": [{
    \\"kind\\": \\"property\\",
    \\"optional\\": false,
    \\"key\\": {
      \\"kind\\": \\"id\\",
      \\"name\\": \\"foo\\"
    },
    \\"value\\": {
      \\"kind\\": \\"boolean\\"
    }
  }],
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;

exports[`typescript named 1`] = `
"export class SomeComponent extends React.Component<{
  foo: boolean;
}> {}
SomeComponent.___types = {
  \\"kind\\": \\"object\\",
  \\"members\\": [{
    \\"kind\\": \\"property\\",
    \\"optional\\": false,
    \\"key\\": {
      \\"kind\\": \\"id\\",
      \\"name\\": \\"foo\\"
    },
    \\"value\\": {
      \\"kind\\": \\"boolean\\"
    }
  }],
  \\"name\\": {
    \\"kind\\": \\"id\\",
    \\"name\\": \\"SomeComponent\\",
    \\"type\\": null
  }
};"
`;
